### PR TITLE
Bookmarks Issue Fixed (#1081)

### PR DESCRIPTION
**Pull request by TejasBhitle**
Fixes issue #1081 

Changes: 
New Interface Added in package org.fossasia.openevent.utils
In Sessions Adapter,  Listener setting method is added.

@the-dagger @mariobehling Please review

### CODE REVIEW

**Comment by the-dagger** [comment]
Thanks,
But there seem to be conflicts due to merging #1113, can you please fix it?
Cheers.

**Comment by free4murad** [comment]
@TejasBhitle 
Don't include `android/build.gradle` in your commit.

**Comment by the-dagger** [comment]
@TejasBhitle any update on this?

**Comment by TejasBhitle** [comment]
@the-dagger Sorry for the delay. My exams are on. I have almost done the work. Will submit it by monday.

File: android/build.gradle [hunk 1]
```diff
@@ -5,7 +5,7 @@ buildscript {
         jcenter()
     }
     dependencies {
-        classpath 'com.android.tools.build:gradle:2.2.2'
+        classpath 'com.android.tools.build:gradle:2.2.3'
```
Comment (harshitagupta30) [inline android/build.gradle]: Can you please revert it back to 2.2.2 ? It is not required.

File: android/app/src/main/java/org/fossasia/openevent/adapters/SessionsListAdapter.java [hunk 2]
```diff
@@ -57,6 +58,11 @@
     public static int listPosition;
     private ColorGenerator colorGenerator = ColorGenerator.MATERIAL;
     private TextDrawable.IBuilder drawableBuilder = TextDrawable.builder().round();
+    private BookmarksListChangeListener bookmarksListChangeListener;
+
+    public void setBookmarksListChangeListener(BookmarksListChangeListener bookmarksListChangeListener){
```
Comment (harshitagupta30) [inline android/app/src/main/java/org/fossasia/openevent/adapters/SessionsListAdapter.java]: Reformat the code.

File: android/app/src/main/java/org/fossasia/openevent/adapters/SessionsListAdapter.java [hunk 3]
```diff
@@ -148,13 +154,15 @@ public void onClick(View v) {
                 if (dbSingleton.isBookmarked(session.getId())) {
                     dbSingleton.deleteBookmarks(session.getId());
                     holder.sessionBookmarkIcon.setImageResource(R.drawable.ic_bookmark_border_white_24dp);
+                    if(bookmarksListChangeListener != null) bookmarksListChangeListener.onChange();
```
Comment (harshitagupta30) [inline android/app/src/main/java/org/fossasia/openevent/adapters/SessionsListAdapter.java]: Same reformat the statement. Also, line 165.

File: android/app/src/main/java/org/fossasia/openevent/adapters/SessionsListAdapter.java [hunk 4]
```diff
@@ -148,13 +154,19 @@ public void onClick(View v) {
                 if (dbSingleton.isBookmarked(session.getId())) {
                     dbSingleton.deleteBookmarks(session.getId());
                     holder.sessionBookmarkIcon.setImageResource(R.drawable.ic_bookmark_border_white_24dp);
+                    if(bookmarksListChangeListener != null){
```
Comment (harshitagupta30) [inline android/app/src/main/java/org/fossasia/openevent/adapters/SessionsListAdapter.java]: reformat if condition.

File: android/app/src/main/java/org/fossasia/openevent/adapters/SessionsListAdapter.java [hunk 5]
```diff
@@ -148,13 +154,19 @@ public void onClick(View v) {
...
+                        bookmarksListChangeListener.onChange();
+                    }
                     Snackbar.make(v, R.string.removed_bookmark, Snackbar.LENGTH_LONG)
                             .setAction(R.string.undo, new View.OnClickListener() {
                                 @Override
                                 public void onClick(View v) {
                                     dbSingleton.addBookmarks(session.getId());
                                     holder.sessionBookmarkIcon.setImageResource(R.drawable.ic_bookmark_white_24dp);
                                     WidgetUpdater.updateWidget(context);
+                                    if(bookmarksListChangeListener != null){
```
Comment (harshitagupta30) [inline android/app/src/main/java/org/fossasia/openevent/adapters/SessionsListAdapter.java]: Reformat the line.

**Comment by mananwason** [comment]
@TejasBhitle Is this good to go?

**Comment by TejasBhitle** [comment]
@mananwason Yes

**Comment by mananwason** [comment]
Cool. Merging then

**Pull request merged**